elisp/gilded-rose.el

```diff
@@ -0,0 +1,37 @@
+(defun make-item (name sell-in quality)
+  "Create an item with NAME, SELL-IN, and QUALITY."
+  (list :name name :sell-in sell-in :quality quality))
+
+(defun update-quality (items)
+  (dolist (item items)
+    (if (and (not (string= (plist-get (car item) :name) "Aged Brie"))
+             (not (string= (plist-get (car item) :name) "Backstage passes to a TAFKAL80ETC concert")))
+
+        (when (> (plist-get (car item) :quality) 0)
+          (when (not (string= (plist-get (car item) :name) "Sulfuras, Hand of Ragnaros"))
+            (setf (plist-get (car item) :quality) (1- (plist-get (car item) :quality)))))
```

Comment:
Is (1- .) a function which decrements by one? Has to be

Reply:
Yes it is:
```
1- is a built-in function in ‘C source code’.

(1- NUMBER)

Return NUMBER minus one.  NUMBER may be a number or a marker.
Markers are converted to integers.

  Other relevant functions are documented in the number group.
  This function does not change global state, including the match
    data.
  This function has a ‘byte-compile’ property ‘byte-compile-one-arg’.
    See the manual for details.
```